### ISSUE TITLE
`eslint-plugin-import` rule `import/no-unused-modules` does not work with vscode-eslint.

### ISSUE THREAD

**Issue by ehmicky** [microsoft/vscode-eslint#717]
The plugin [`eslint-plugin-import`](https://github.com/benmosher/eslint-plugin-import) has a rule [`import/no-unused-modules`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-unused-modules.md) that checks (among other things) for exports that are not used in other files.

```json
"import/no-unused-modules": [2, {"unusedExports": true}]
```

This does not work well with vscode-eslint: files that have not been opened in the editor yet are not taken into account. As a consequence exports are being reported as unused even though they are used.

For example, if I just open a new VSCode Window:

![Screenshot from 2019-07-18 12-43-27](https://user-images.githubusercontent.com/8136211/61451485-c68fb100-a959-11e9-8582-18433b1431f5.png)

The `serialize()` exported function is reported, even after saving the file.

Now if I open that other file which imports the `serialize()` function:

![Screenshot from 2019-07-18 12-44-04](https://user-images.githubusercontent.com/8136211/61451498-cbecfb80-a959-11e9-91c0-69f98091e1aa.png)

Then go back to the first file and save it, the reporting is now gone:

![Screenshot from 2019-07-18 12-44-11](https://user-images.githubusercontent.com/8136211/61451505-cf808280-a959-11e9-990e-1e9caf165573.png)

It seems like only files that are open within VSCode are considered for the `import/no-unused-modules` rule.

This behavior only seems to happen with VSCode. If I run `eslint` on the command line, everything works fine.

I've initially opened [an issue](https://github.com/benmosher/eslint-plugin-import/issues/1422) with `eslint-plugin-import` but got redirected here by its author.

VSCode: `1.36.1`
vscode-eslint: `1.9.0`
OS: Ubuntu `19.04`
ESLint: `6.0.1`
eslint-plugin-import: `2.18.0`	
Node: `12.6.0`

**Comment by dbaeumer** [comment]
@ehmicky ESLint only validates the open files. This is comparable to running eslint in the terminal for a single file. E.g. `eslint folder/file.js`. If you do this are the exports reported as well?

There is also a eslint validate task that validates the whole project. It can be enable with `"eslint.provideLintTask": true`. You can invoke the task using `Run Task` from the Terminal menu.

**Comment by ehmicky** [comment]
@dbaeumer thanks for looking into this.

If I understand you correctly, VSCode-ESLint only validates the opened files. This would be an issue with rules like [`no-unused-modules`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-unused-modules.md) where linting needs to happen project-wise and not file-wise. 

Doing `eslint folder/file.js` works (i.e. the exports are not reported) because that ESLint rule receives a `src` option (defaulting to current directory) and manually recurse over it to find out which files are depending on `folder/file.js`. At least that's what I think the rule does under the hood, I am not completely sure ([source code](https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/no-unused-modules.js)). It just works (outside of VSCode-ESLint).

The `eslint.provideLintTask` would unfortunately require an explicit action to be triggered, as opposed to linting in realtime.

**Comment by dbaeumer** [comment]
@ehmicky thanks for the additional information. The problem is that we can't do that on typing since it would not scale parsing all files in a workspace.

What would work is that we auto trigger the lint task on save so that there is not need for a user interaction.

**Comment by dbaeumer** [comment]
Good opportunity for a PR.

**Comment by ehmicky** [comment]
@dbaeumer yes triggering the lint task on save would work for me, thanks!

**Comment by dbaeumer** [comment]
I am still happy to accept a PR if someone wants to work on this.

**Comment by AndysonDK** [comment]
I get the same error would love to see a PR

**Comment by adamsuskin** [comment]
I am seeing this issue manifest with two open files. ESLint on the command-line works fine.

VSCode: `1.65.2`
vscode-eslint: `2.2.2`
OS: macOS Monterey `12.3`
ESLint: `8.4.1`
eslint-plugin-import: `2.25.4`
Node: `16.14.0`
